web/styles/left_sidebar.css

```diff
@@ -224,14 +219,115 @@ li.show-more-topics {
     }
 }
 
-.narrows_panel {
-    & li a {
-        margin-top: 1px;
+#views-label-container {
+    cursor: pointer;
+    /* Padding of #private_messages_section_header. */
+    padding: 0 10px 1px 6px;
+    /* The margin-left of .private_messages_container. */
+    margin-left: 6px;
+    /* Space between #views-label-container and #left-sidebar-navigation-list in collapsed state. */
+    margin-right: 10px;
+    margin-top: 3px;
+    outline: none;
+}
 
-        &:hover {
+#left-sidebar-navigation-area {
+    & li.top_left_row {
+        position: relative;
+        padding: 2px 10px 1px $far_left_gutter_size;
+
+        & a:hover {
             text-decoration: none;
         }
     }
+
+    &.collapsed {
+        display: flex;
+        align-items: center;
+        /* 2px space between views and direct messages sections when views section is collapsed. */
+        margin-bottom: 2px;
+
+        /* Only show first 5 views in collapsed state. */
+        .top_left_row:nth-child(n + 6) {
+            display: none !important;
+        }
+
+        & #left-sidebar-navigation-list {
+            width: 100%;
+            display: flex;
+            /* 25px margin-right of .subscription_block + 12px margin-right of #stream_filters. */
+            margin-right: 37px;
+            margin-bottom: 0;
+            padding: 0;
+
+            & li {
```

Comment:
Let's use the `top_left_row` class for now (I think it and `bottom_left_row` should be rewritten during the redesign, but let's worry about that then).